CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+# Contribution to CryptoAnaylsis
+You can contribute with issues and PRs. Simply filing issues for problems you encounter is a great 
+way to contribute. Contributing implementations is greatly appreciated.
+
+## DOs and DON'Ts
+Please do:
+- **DO** follow the [coding guidelines](CODING.md)
+- **DO** follow the naming conventions for branches.
+- **DO** include tests when adding new features. When fixing bugs, start with adding a test that highlights how the current behavior is broken.
+- **DO** keep the discussions focused. When a new or related topic comes up it's often better to create new issue than to side track the discussion.
+- **DO** document your changes in detail in the pull requests.
+
+Please do not:
+- **DON'T** make PRs for coding guidelines changes.
+- **DON'T** create a big pull requests with no related issue. Instead, first file an issue.
+- **DON'T** submit PRs that use licensed files we are not allowed to use.
+- **DON'T** add or change API related code without filing an issue and discussing with the project leads first
+
+## Suggested Workflow
+
+We aim for the following workflow:
+
+1. Create an issue for your work.
+  - You can skip this step for trivial changes.
+  - Reuse an existing issue on the topic, if there is one.
+  - If you want to implement the issue yourself, please state this. 
+2. Create a personal fork of the repository on GitHub. 
+3. In your fork, create a new branch of develop (`git checkout -b mybranch`).
+    - Apply branch naming rules (see above)
+3. Make and commit your changes to your branch.
+    - Keep changes as small as possible.
+    - Try to follow the [Commit Messages](#commit-messages) guidance.
+4. Add new tests corresponding to your change, if applicable.
+5. Build the repository with your changes.
+    - Make sure that the builds are clean.
+    - Make sure that the tests are all passing, including your new tests.
+6. Create a pull request against this repository's `develop` branch.
+    - If the branch is not yet ready, give the PR's name a `WIP` prefix. 
+    - Check if all the Continuous Integration checks are passing.
+7. When you are done, add one or more reviewer.
+    - Make sure the PR has the latest changes from `develop` merged in it.
+8. Wait for feedback or approval of your changes.
+9. When remaining issues from feedback are resolved and all checks are green, your PR will be merged by one of the reviewers.
+    - Delete the source branch with the merge.
+
+
+## Branches
+
+This repository contains two central branches. The **master** and the **develop** branch. The develop branch is default.
+Both branches are *protected* against direct write access thus Pull Requests are necessary to push into them. 
+Other branches are unprotected and can be created and deleted by an contributer
+
+The `master` branch holds the lastest stable release of the application.
+
+The `develop` branch holds the lastest development version of the application
```

Comment:
I would add one more line on the expected workflow. For example, how and when is development synchronized into master.

Reply:
That's something we have not yet decided